internal/engine/ast.go

```diff
@@ -357,19 +383,43 @@ func buildExpr(expr *exprpb.Expr, acc *enginev1.PlanResourcesFilter_Expression_O
 		}
 	case *exprpb.Expr_StructExpr:
 		x := expr.StructExpr
+		if c := parent.GetCallExpr(); c != nil {
+			// convert struct to a list, if it is a key membership test operation
+			const nArgs = 2
+			const rhsIndex = 1 // right-hand side arg index
+			if c.Function == operators.In && len(c.Args) == nArgs && c.Args[rhsIndex] == cur {
```

Comment:
Are there other functions similar to `in` that we should be handling here?

Reply:
Good catch. From the top of my head, other map-related functions/macros include `all` and `exists_one`, which can be transformed similarly.

Reply:
I prefer to implement them in a separate PR.